Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs

```diff
@@ -80,35 +88,38 @@ private void UpdateUiState(Entity<ReagentDispenserComponent> reagentDispenser)
             return null;
         }
 
-        private List<ReagentId> GetInventory(Entity<ReagentDispenserComponent> ent)
+        private List<KeyValuePair<string, KeyValuePair<string, string>>> GetInventory(ReagentDispenserComponent reagentDispenser)
         {
-            var reagentDispenser = ent.Comp;
-            var inventory = new List<ReagentId>();
+            var inventory = new List<KeyValuePair<string, KeyValuePair<string, string>>>();
 
-            if (reagentDispenser.PackPrototypeId is not null
-                && _prototypeManager.TryIndex(reagentDispenser.PackPrototypeId, out ReagentDispenserInventoryPrototype? packPrototype))
+            for (var i = 0; i < reagentDispenser.NumSlots; i++)
             {
-                inventory.AddRange(packPrototype.Inventory.Select(x => new ReagentId(x, null)));
-            }
+                var storageSlotId = ReagentDispenserComponent.BaseStorageSlotId + i;
+                var storedContainer = _itemSlotsSystem.GetItemOrNull(reagentDispenser.Owner, storageSlotId);
+
+                // Set label from manually-applied label, or metadata if unavailable
+                string reagentLabel;
+                if (EntityManager.TryGetComponent<LabelComponent?>(storedContainer, out var label) && label.CurrentLabel != null)
+                    reagentLabel = label.CurrentLabel;
+                else if (EntityManager.TryGetComponent<MetaDataComponent?>(storedContainer, out var metadata))
```

Comment:
use TryComp instead of EntityManager.TryGetComponent in all the places you have it.